src/Shortcodes/ImageShortcodeProvider.php

```diff
@@ -84,18 +84,18 @@ public static function handle_shortcode($args, $content, $parser, $shortcode, $e
 
         // Build the HTML tag
         $attrs = array_merge(
-            // Set overrideable defaults
-            ['src' => '', 'alt' => $record->Title],
+            // Set overrideable defaults ('alt' must be present regardless of contents)
+            ['src' => '', 'alt' => ''],
             // Use all other shortcode arguments
             $args,
             // But enforce some values
             ['id' => '', 'src' => $src]
         );
 
-        // Clean out any empty attributes
-        $attrs = array_filter($attrs, function ($v) {
-            return (bool)$v;
-        });
+        // Clean out any empty attributes (aside from alt)
+        $attrs = array_filter($attrs, function ($k, $v) {
+            return (bool)$v || $k === 'alt';
```

Comment:
should we be checking `strlen(trim($v))` rather than `(bool)$v` in here? It feels we don't really want php semantics in here.

Reply:
`!empty($v)` more accurately tracks the existing behaviour, so I might use that instead? Though your suggestion seems a bit more specific and sensible in this context.

Reply:
though I reckon the current behaviour is incorrect. E.g. `"0"` is a valid value for an html attribute

Reply:
Your change wouldn't stop `"0"` from passing this filter, though - and I'd say that's more the work of the HTML parser?

Reply:
```php
php > $v = "0";
php > echo (bool)$v;  // == false
php > echo !empty($v);  // == false
php > echo strlen(trim($v));  // == true
```

Reply:
yeah, my point is we shouldn't be deleting `"0"` from the array

Reply:
Gotcha - I'll tweak it.